client-tools/Restore/RestoreFeature.cpp

```diff
@@ -168,6 +168,44 @@ uint64_t getReplicationFactor(arangodb::RestoreFeature::Options const& options,
   return result;
 }
 
+/// @brief return the target replication factor for the specified collection
+uint64_t getWriteConcern(arangodb::RestoreFeature::Options const& options,
+                         arangodb::velocypack::Slice const& slice) {
+  uint64_t result = 1;
+
+  arangodb::velocypack::Slice s =
+      slice.get(arangodb::StaticStrings::WriteConcern);
+  if (s.isInteger()) {
+    result = s.getNumericValue<uint64_t>();
+  }
+
+  s = slice.get("name");
+  if (!s.isString()) {
+    // should not happen, but anyway, let's be safe here
+    return result;
+  }
+
+  if (!options.writeConcern.empty()) {
+    std::string const name = s.copyString();
+
+    for (auto const& it : options.writeConcern) {
+      auto parts = arangodb::basics::StringUtils::split(it, '=');
+      if (parts.size() == 1) {
+        result = arangodb::basics::StringUtils::uint64(parts[0]);
+      }
+
+      if (parts.size() != 2 || parts[0] != name) {
```

Comment:
The second `if` condition looks wrong, because if `parts.size() == 1` evaluates to `true` in line 193, we fetch the numeric result in line 194, but then immediately consider it invalid in line 197.
I guess the two `if`s here should be combined by an `else`.
```suggestion
      if (parts.size() == 1) {
        result = arangodb::basics::StringUtils::uint64(parts[0]);
      } else if (parts.size() != 2 || parts[0] != name) {
```

Reply:
Or is this actually intentional, so that we pick up the value in line 194, but keep looking for if there is a better, more precise match?

Reply:
Yes, similar to `replication-factor` we can override for all collection with `--replication-factor 2` but then add this multiple times to handle specific collections differently `--replication-factor foo=3`.